stockroom/external/importer/torchvision_importers.py

```diff
@@ -81,3 +86,325 @@ class FashionMnist(TorchvisionCommon):
+        utils.download(urls, root)
+        files = [os.path.join(root, file) for file in urls]
+        utils.unzip(files, root)
+
+
+class COCOCaptions(BaseImporter):
+    name = "coco_captions"
+
+    def __init__(self, root, ann_file, split):
+        self.dataset = datasets.CocoCaptions(root, ann_file)
+        self.split = split
+        self.sample_img, self.sample_captions = self._process_data(*self.dataset[0])
+
+    def column_names(self):
+        return f"{self.name}-{self.split}-image", f"{self.name}-{self.split}-captions"
+
+    def shapes(self):
+        return (3, 640, 640), None
+
+    def variability_status(self):
+        return True, False
+
+    def dtypes(self):
+        return self.sample_img.dtype, ["str"]
+
+    def __iter__(self):
+        for img, captions in self.dataset:
+            yield self._process_data(img, captions)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    @staticmethod
+    def _process_data(img, captions):
+        img = np.transpose(np.array(img), (2, 0, 1))
+        img = np.ascontiguousarray(img)
+        captions = {i: cap for i, cap in enumerate(captions)}
+
+        return img, captions
+
+    @classmethod
+    def gen_splits(cls, root):
+        root = Path(root)
+        setup_coco_file(root)
+
+        # files for training
+        train_imgs = root / "train2017"
+        val_imgs = root / "val2017"
+
+        train_annotations = root / "annotations/captions_train2017.json"
+        val_annotations = root / "annotations/captions_val2017.json"
+
+        return [
+            cls(train_imgs, train_annotations, "train"),
+            cls(val_imgs, val_annotations, "val"),
+        ]
+
+
+class COCODetection(BaseImporter):
+    name = "coco_detection"
+
+    def __init__(self, root, ann_file, split):
+        self.dataset = datasets.CocoDetection(root, ann_file)
+        self.split = split
+        self.sample_img, self.sample_boxes = self._process_data(*self.dataset[0])
+
+    def column_names(self):
+        return f"{self.name}-{self.split}-image", f"{self.name}-{self.split}-boxes"
+
+    def shapes(self):
+        _, box = list(self.sample_boxes.items())[0]
+        return (3, 640, 640), box.shape
+
+    def variability_status(self):
+        return True, False
+
+    def dtypes(self):
+        _, box = list(self.sample_boxes.items())[0]
+        return self.sample_img.dtype, [box.dtype]
+
+    def __iter__(self):
+        for img, boxes in self.dataset:
+            yield self._process_data(img, boxes)
+
+    def __len__(self):
+        return len(self.dataset)
+
+    @staticmethod
+    def _process_data(img, boxes):
+        img = np.transpose(np.array(img), (2, 0, 1))
+        img = np.ascontiguousarray(img)
+
+        boxes = {
+            box["category_id"]: np.array(box["bbox"], dtype=np.float64) for box in boxes
+        }
+
+        return img, boxes
+
+    @classmethod
+    def gen_splits(cls, root):
+        root = Path(root)
+        setup_coco_file(root)
```

Comment:
if my understanding is correct, `boxes`  will have multiple boxes with same `category_id`. Saving with this structure will fail

Reply:
Yeah! that was a big blunder from my part.
fixing it.